#! /usr/bin/env python

PACKAGE='aero_ssd_recognition'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gpu", int_t, 0, "Use GPU or CPU", 0, -1, 10)
gen.add("model_name", str_t, 0, "Name of SSD model to use", None)
gen.add("score_thresh", double_t, 0, "Detection Threshold", 0.6, 0.0, 1.0)
gen.add("eco_mode", bool_t, 0, "Eco mode flag", False)
gen.add("eco_hz", double_t, 0, "eco mode frame rate", 1.0, 0.1, 5.0)
gen.add("boost_hz", double_t, 0, "boost mode frame rate", 15.0, 1.0, 30.0)
gen.add("visualize", bool_t, 0, "Visualization", False)

exit(gen.generate(PACKAGE, "aero_ssd_recognition", "ObjectDetector"))
